### PR TITLE
Fix incorrect assertion in Java concat [skip ci]

### LINKED ISSUE #8246: [BUG] incorrect assertion in stringConcatenate and listConcatenateByRow
**Describe the bug**
There exists incorrect assertion on length of input columns in both [stringConcatenate](https://github.com/rapidsai/cudf/blob/92f067dcd80ce11c0431432f3817bebfa97816a0/java/src/main/java/ai/rapids/cudf/ColumnVector.java#L512) and [listConcatenateByRow](https://github.com/rapidsai/cudf/blob/92f067dcd80ce11c0431432f3817bebfa97816a0/java/src/main/java/ai/rapids/cudf/ColumnVector.java#L551). However, libcudf supports single column concatenation for both string and list_by_row. And single column concatenation is also valid in Spark.

**Expected behavior**
We should reduce the requirement of length from 2 to 1, so as to unblock single column inputs from Spark rapids.

### PR DESCRIPTION
**Pull request by sperlingxx**
Fix #8246.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -509,7 +509,8 @@ public static ColumnVector stringConcatenate(ColumnView[] columns) {
    * @return A new java column vector containing the concatenated strings.
```
Comment (GaryShen2008) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java:509]: Need to update the comment here to match the behavior.

Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java:509]: Fixed! Thanks for catching that.

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 2]
```diff
@@ -2085,6 +2084,16 @@ void testStringConcatWithNulls() {
          ColumnVector concat = ColumnVector.stringConcatenate(emptyString, nullSubstitute, new ColumnView[]{v, v})) {
       assertColumnsAreEqual(concat, e_concat);
     }
+
+    try (ColumnVector v = ColumnVector.fromStrings("a", "B", "cd", "\u0480\u0481", "E\tf",
+        "g\nH", "IJ\"\u0100\u0101\u0500\u0501",
+        "kl m", "Nop1", "\\qRs2", null,
+        "3tuV\'", "wX4Yz", "\ud720\ud721");
+         Scalar emptyString = Scalar.fromString("");
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:2092]: @wbo4958  For this single column case, the `separator` affects nothing.

**Comment by firestarman** [comment]
LGTM from the functionality point.
For perf, it is recommended to return the single column directly instead of this `concatenate` op.

**Review by firestarman — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
rerun tests

**Review by revans2 — APPROVED** [review]

**Comment by sperlingxx** [comment]
@gpucibot merge

**Pull request merged**